src/goCheck.ts

```diff
@@ -58,6 +60,19 @@ export function check(fileUri: vscode.Uri, goConfig: vscode.WorkspaceConfigurati
 	const runningToolsPromises = [];
 	const cwd = path.dirname(fileUri.fsPath);
 	const goRuntimePath = getBinPath('go');
+	const languageServerTool = getToolFromToolPath(getLanguageServerToolPath());
+	const languageServerOptions: any = goConfig.get('languageServerExperimentalFeatures');
+	let languageServerFlags: string[] = goConfig.get('languageServerFlags');
+	if (!Array.isArray(languageServerFlags)) {
+		languageServerFlags = [];
+	}
+
+	let disableBuild = languageServerOptions['diagnostics'] === true && (languageServerTool === 'gopls' || languageServerTool === 'bingo');
+	const disableVet = languageServerOptions['diagnostics'] === true && languageServerTool === 'gopls';
+
+	if (disableBuild && languageServerTool === 'bingo' && languageServerFlags.indexOf('-diagnostics-style=none') > -1) {
```

Comment:
>disableBuild [](start = 5, length = 12)

this condition seems weird. if ( disableBuild) disableBuild  = true ?

Reply:
good catch, it should be set to false